OpenSim/Simulation/Model/GeometryPath.h

```diff
@@ -47,202 +44,350 @@ class ScaleSet;
+    void applyWrapObjects(const SimTK::State&, Array<const AbstractPathPoint*>&) const;
 
-private:
+    double calcLengtheningSpeed(const SimTK::State&, const Array<const AbstractPathPoint*>&) const;
 
-    void computePath(const SimTK::State& s ) const;
-    void computeLengtheningSpeed(const SimTK::State& s) const;
-    void applyWrapObjects(const SimTK::State& s, Array<AbstractPathPoint*>& path ) const;
-    double calcPathLengthChange(const SimTK::State& s, const WrapObject& wo, 
-                                const WrapResult& wr, 
-                                const Array<AbstractPathPoint*>& path) const; 
-    double calcLengthAfterPathComputation
-       (const SimTK::State& s, const Array<AbstractPathPoint*>& currentPath) const;
+    double calcPathLengthChange(
+        const SimTK::State&,
+        const WrapObject&,
+        const WrapResult&,
+        const Array<const AbstractPathPoint*>&) const;
+
+    double calcPathLength(
+        const SimTK::State&,
+        const Array<const AbstractPathPoint*>&) const;
 
-    void constructProperties();
     void namePathPoints(int aStartingIndex);
-    void placeNewPathPoint(const SimTK::State& s, SimTK::Vec3& aOffset, 
-                           int index, const PhysicalFrame& frame);
-    //--------------------------------------------------------------------------
-    // Implement Object interface.
-    //--------------------------------------------------------------------------
-    /** Override of the default implementation to account for versioning. */
-    void updateFromXMLNode(SimTK::Xml::Element& aNode, int versionNumber = -1) override;
+
+    void placeNewPathPoint(
+        const SimTK::State&,
+        SimTK::Vec3& aOffset,
+        int index,
+        const PhysicalFrame&);
+
+    //=============================================================================
+    // DATA
+    //=============================================================================
+
+    // used for scaling tendon and fiber lengths
+    double _preScaleLength;
+
+    // solver used to compute moment-arms
+    //
+    // the GeometryPath owns this object, but it is cleared whenever the GeometryPath
+    // is copied
+    SimTK::ResetOnCopy<std::unique_ptr<MomentArmSolver>> _maSolver;
+
+    // internal cache of raw pointers to the path's points
+    //
+    // this is here for legacy reasons: the path points are *actually* held in
+    // a cache variable as `OpenSim::ComponentPath`s. However, existing APIs might
+    // want an array of raw pointers.
+    //
+    // reset on copy to prevent aliasing issues.
+    mutable SimTK::ResetOnCopy<Array<const AbstractPathPoint*>> _currentPathPtrsCache;
+
+    // cache variable for the path's length
+    mutable CacheVariable<double> _lengthCV;
+
+    // cache variable for the path's lengthening speed
+    mutable CacheVariable<double> _speedCV;
+
+    // cache variable for the (order of) the path's points
```

Comment:
This information could be packed into 32-bits, rather than 64, but that's a nice-to-have extra step